app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/EventsController.java

```diff
@@ -34,115 +32,49 @@
     public EventsController(Context context) {
         this.context = context;
         eventDao = TcaDb.getInstance(context).eventDao();
-        ticketDao = TcaDb.getInstance(context).ticketDao();
-        ticketTypeDao = TcaDb.getInstance(context).ticketTypeDao();
+        downloadFromService();
     }
 
 
-    public void downloadFromService(boolean force) {
-        TUMCabeClient api = TUMCabeClient.getInstance(context);
+    public void downloadFromService() {
+        // get ticket type information from API
+        Thread thread = new Thread(){
```

Comment:
No, please use Retrofit with callbacks. Never directly start a new thread please.

Reply:
Ok, I see. I implemented a retrofit callback. Please check if I did it correctly...